Fix dependency installation issues

cypheroxide opened the pull request:
# Fix dependency installation issues

## Issues Encountered

While trying to set up the AI-Writer project across multiple environments (Ubuntu 22.04, PopOS, Arch Linux, Windows 10, Windows 11), I consistently ran into dependency installation failures that weren't documented in the setup instructions. These issues included:

1. Build failures for packages that require C/C++ compilation due to missing Visual C++ Build Tools on Windows
2. Failed installations of `tiktoken` and related packages due to missing Rust compiler
3. Cascade of dependency conflicts where resolving one issue would trigger another
4. Lack of clear information about system prerequisites in the documentation

These issues made it extremely difficult for new users to get started with the project, as evidenced by multiple reopened issues on GitHub related to installation problems.

## Changes Made

This PR implements several improvements to address these installation issues:

1. **Enhanced documentation**: Added a comprehensive "Prerequisites" section to the README that clearly lists all required system dependencies for both Windows and Linux environments.

2. **Installation helper script**: Created `install_dependencies.py` that:
   - Automatically checks for required system dependencies (Python version, Visual C++ Build Tools, Rust compiler)
   - Provides clear, actionable guidance for installing any missing components
   - Offers to set up the virtual environment and install Python dependencies when all prerequisites are met

3. **Explicit installation steps**: Updated the installation instructions with clearer, more detailed steps that account for potential issues

## How These Changes Address the Issues

These improvements directly address the installation challenges by:

1. **Preventing problems before they occur**: By clearly documenting prerequisites and providing automated checks, users will install necessary system components before attempting to install Python dependencies.

2. **Providing clear guidance**: When issues do occur, users now have specific instructions for resolving them rather than generic error messages.

3. **Reducing support burden**: These changes should significantly reduce the number of "cannot install" issues, allowing more focus on feature development.

4. **Improving user experience**: New users will have a much smoother onboarding experience, increasing adoption and community engagement.

5. **Ease of installation for new users**: As many new users seem to attempt to use this platform on fresh Windows systems (and some Linux systems), this reduces the burden of coding and debugging knowledge, increasing adoption and streamlining installation processes.

## Testing

I've tested these and confirmed changes across multiple environments:
- Windows 11 with Python 3.12.9
- Ubuntu 22.04 with Python 3.13.2

In all cases, following the new instructions and using the helper script resulted in successful installation and running of the application.

## Related Issues

This PR addresses the issues described in [Issue #142 ](https://github.com/AJaySi/AI-Writer/issues/142) regarding dependency installation problems.

AJaySi commented:
@cypheroxide These are great enhacements.
I will code review the changes soon and update here. @uniqueumesh Please help review.

Thank you so much for your efforts.

AJaySi commented on install_dependencies.py:
```diff
@@ -0,0 +1,111 @@
+        print("\nAll system dependencies found!")
+        
+        # Ask user if they want to proceed with installation
+        response = input("\nWould you like to create a virtual environment and install Python dependencies? (y/n): ")
```
Awesome.

AJaySi commented on install_dependencies.py:
```diff
@@ -0,0 +1,111 @@
+    print("\nVisual C++ Build Tools are required to build certain Python packages.")
+    print("To install Visual C++ Build Tools:")
+    print("Option 1: Run this command in an administrative PowerShell:")
+    print("  winget install Microsoft.VisualStudio.2022.BuildTools --silent --override \"--wait --quiet --add Microsoft.VisualStudio.Workload.VCTools --includeRecommended\"")
```
Very informative messages. Thanks. Lets log them, so when they fail we will have logfile to debug with.

cypheroxide replied:
Ill look into adding a logging function for error tracking.

AJaySi commented on README.md:
```diff
@@ -46,6 +46,22 @@ By bringing these powerful solutions together, Alwrity ensures a streamlined wor
+- Microsoft Visual C++ Build Tools 14.0 or greater
+  - Install with: `winget install Microsoft.VisualStudio.2022.BuildTools --silent --override "--wait --quiet --add Microsoft.VisualStudio.Workload.VCTools --includeRecommended"`
+- Rust Compiler
+  - Install with: `Invoke-WebRequest -Uri https://static.rust-lang.org/rustup/dist/x86_64-pc-windows-msvc/rustup-init.exe -OutFile rustup-init.exe; ./rustup-init.exe -y`
```
Mention install_dependencies script here, so users can use that instead of manual install.

cypheroxide replied:
I'll add a note that there's an install script for the dependencies.

cypheroxide commented:
Added a section to `README.md` to mention the `install_dependencies.py` script for ease of installation to run the app. I also updated the `install_dependencies.py` script to include better logging of failures during dependency installation.

AJaySi reviewed (approved):
All Great. Thank you so much. This is great addition and will help greatly in tool adoption.
Regards and Respect.

AJaySi reviewed (approved):
Looks good to me.

AJaySi reviewed (approved):
Looks Good,

AJaySi commented:
@cypheroxide 

Hello Angel, 
I have approved from my end and all checks done. 
I think you made changes on an older local repo, while the github branch got updated. Hence, the conflicts.

**How To Fix: This branch cannot be rebased due to conflicts**

- Ensure our branch is up to date with the main branch(Rebase).
- Reset the local branch to the main branch - without altering the working copy.
- Create a new commit to represent the merge. ...
- Finally, push the your remote branch.

<img width="711" alt="image" src="https://github.com/user-attachments/assets/6223d476-580c-4277-989f-5f909d6391fc" />

cypheroxide commented:
Oops. It's been a long week at work, so that probably did a thing to my brain. I'll fix that and update the PR as necessary.

cypheroxide commented:
I _think_ I fixed the issue.  I'm not sure what happened, but it looks like I was able to rebase the branch and resolve the conflicts.  It should be good, now.

AJaySi commented:
**Pull request successfully merged and closed**

@cypheroxide : Done. thanks a lot. Awesome.
(Just pushing my luck): So, whats the next thing you will contribute to ? Please, Please, pretty please ....

cypheroxide commented:
> So, whats the next thing you will contribute to ? Please, Please, pretty please ....

### **Short answer:**
I'm not sure at the moment. We had a major update at work and I have a few things to focus on there, so may be a little less bug hunting, PRs, and dev work, at least until I make progress there.

### **Better answer:**
Whatever catches my fancy and I find an issue that I might be able to resolve. My background is predominantly CySec and ML/AI workflows, and that ties in with my job. If I come across something where I could help, I'll join the conversation.

AJaySi commented:
Sorry, I didnt mean to impose and i cant. It was just a humble request. 
Alwrity needs people like you, so we build something against $/month subscriptions.

We are already grateful to you for your contributions and being greedy asking for more.
Please stick around and chime in to whatever catches your fantasy.

Regards.
